common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -355,8 +441,31 @@ void closeAndDeletePartitionFiles(Map<Integer, AppShufflePartitionInfo> partitio
       .forEach((partitionId, partitionInfo) -> {
         synchronized (partitionInfo) {
           partitionInfo.closeAllFilesAndDeleteIfNeeded(true);
+          cleanUpAppShufflePartitionInfoInDB(partitionInfo.appShufflePartitionId);
         }
       });
+
+  }
+
+  void cleanUpAppShufflePartitionInfoInDB(AppShufflePartitionId appShufflePartitionId) {
+    if (db != null) {
+      try {
+        if (appShufflePartitionId.reduceId == -1) {
+          db.delete(dbAppAttemptShufflePartitionKeyWithCustomPrefix(
+            appShufflePartitionId, APP_ATTEMPT_SHUFFLE_FINALIZE_STATUS_KEY_PREFIX));
+        } else {
+          db.delete(dbAppAttemptShufflePartitionKeyWithCustomPrefix(
+            appShufflePartitionId, APP_ATTEMPT_SHUFFLE_PARTITION_KEY_PREFIX));
+        }
+      } catch (IOException e) {
+        logger.error("Error deleting {}_{} shuffleMergeId {} shuffle {}"
+                + "reduce {} from application shuffle merged partition info db",
+            appShufflePartitionId.appId, appShufflePartitionId.attemptId,
+            appShufflePartitionId.shuffleMergeId,
+            appShufflePartitionId.shuffleId,
```

Comment:
`shuffleId` before `shuffleMergeId` (here and in `updateChunkInfo` - not sure if elsewhere as well).
Also, use `toString` in `AppShufflePartitionId` instead (here and elsewhere we are logging) ?